packages/wrangler/src/api/pages/publish.tsx

```diff
@@ -232,24 +246,49 @@ export async function publish({
 	 */
 	if (_workerJS) {
 		let workerFileContents = _workerJS;
-		if (bundle) {
+		const enableBundling = bundle || experimentalWorkerBundle;
+
+		if (enableBundling) {
 			const outfile = join(tmpdir(), `./bundledWorker-${Math.random()}.mjs`);
-			await buildRawWorker({
+			workerBundle = await buildRawWorker({
 				workerScriptPath,
 				outfile,
 				directory: directory ?? ".",
 				local: false,
 				sourcemap: true,
 				watch: false,
 				onEnd: () => {},
+				experimentalWorkerBundle
 			});
 			workerFileContents = readFileSync(outfile, "utf8");
 		} else {
 			await checkRawWorker(workerScriptPath, () => {});
 		}
 
-		formData.append("_worker.js", new File([workerFileContents], "_worker.js"));
-		logger.log(`✨ Uploading _worker.js`);
+		if(experimentalWorkerBundle) {
+			const workerBundleFormData = createWorkerBundleFormData(workerBundle as BundleResult);
+			let workerBundleContents = await new Response(workerBundleFormData).text();
+
+			/**
+			 * <hack-alert>
+			 */
+			const undiciGeneratedBoundary = workerBundleContents.split("\r\n")[0];
+			workerBundleContents = workerBundleContents.replaceAll(undiciGeneratedBoundary, "------pages-generated-boundary-00000000");
```

Comment:
@GregBrimble if we do smth like this 👆, would we still need those 70 bytes thing we talked abt? Unless `undici` changes how they generate the response body from the formData
👉 https://github.com/nodejs/undici/blob/f02ab5b18a7e5cfcf0ffafaf8dc7a88b1f8430e3/lib/fetch/response.js#L156
👉 https://github.com/nodejs/undici/blob/f02ab5b18a7e5cfcf0ffafaf8dc7a88b1f8430e3/lib/fetch/response.js#L476
👉 https://github.com/nodejs/undici/blob/f02ab5b18a7e5cfcf0ffafaf8dc7a88b1f8430e3/lib/fetch/body.js#L31
👉 https://github.com/nodejs/undici/blob/f02ab5b18a7e5cfcf0ffafaf8dc7a88b1f8430e3/lib/fetch/body.js#L108

the first line of this body will contain just the undici-generated boundary. I would assume they are returning a pretty standard format, but I'm pretty new with this stuff, so I don't want to assume anything